### DISCUSSION TITLE
How to implement a "rental" system?

### DISCUSSION THREAD

**Discussion by blu-IT** [YetiForceCompany/YetiForceCRM#17004] [How to]
Hi,

I have some devices that can be rent to customers, each with a unique serial number. Now I am not sure how to get the devices correctly in the YF (assets or products). Important is, that they go aout of the warehouse to the customer and the also come back to the warehouse/stock when the rent period is over.
I also saw that there are assets and products in YF - each with own serial numbers (but different database fields _serialnumber_ and _serial_no_ !). For me it is important, that a unique device can only be rent once at a time (logical ;-) ) and when it is out of the warehouse, it can't be accidently choosen in other modules (out of stock). I saw that an asset can be combined with a product, but I stilll don't get the sense in particular, when both - asset and product - have a serial number.

Can someone explain me these two possibilities or has own experience with a "rental system" in YF? 
Maybe also with already implemented charging/invoicing of this service / service contract?

Thanks!

Kind regards

**Comment by mariuszkrzaczkowski** [comment]
see the module "Products sold" this is the closest to your need
https://gitdeveloper.yetiforce.com/index.php?module=Assets&view=List&mid=88&parent=84
https://gitdeveloper.yetiforce.com/index.php?module=OSSSoldServices&view=List&mid=93&parent=84
